packages/odie-client/src/components/notices/use-message-size-error-notice.tsx

```diff
@@ -0,0 +1,24 @@
+import { __ } from '@wordpress/i18n';
+import { useCallback } from 'react';
+import { useOdieAssistantContext } from '../../context';
+
+export default function useMessageSizeErrorNotice() {
+	const { setNotice } = useOdieAssistantContext();
+
+	const isMessageLengthValid = useCallback(
+		( message?: string ) => {
+			if ( message?.length && message.length > 4096 ) {
+				setNotice(
+					'message-size-error',
+					__( 'Message exceeds 4096 characters limit.', __i18n_text_domain__ )
+				);
+				return false;
+			}
+			setNotice( 'message-size-error', null );
+			return true;
+		},
+		[ setNotice ]
+	);
```

Comment:
I find it confusing that the `isMessageLengthValid` function both checks whether the length is valid and also sets the notice text. I think it'd make more sense to have a hook for the message length, and then a different one for the notice. Wdyt?